### DISCUSSION TITLE
What is the best way to translate customized date and adopt it globally?

### DISCUSSION THREAD

**Discussion by nkwwk** [amannn/next-intl#946] [General]
Hello, community,

I want to ask about the pattern of `25th of December 2023`; it includes the pluralization of the date and the `of`, which seems to require a customized text render in the date format. But when I looked into the dates and times docs, they support `Intl.DateTimeFormat` but are not adaptable for this case. How should I handle a case like this?

**Comment by amannn** [comment]
For the "25th" part, you could potentially achieve that with [ordinal pluralization](https://next-intl-docs.vercel.app/docs/usage/messages#ordinal-pluralization).

But the more general question is how this would look like with other languages. E.g. in German, we'd say "25. Dezember 2023"—so there's nothing like the "of" from the English example. So maybe you're better off choosing a format that is supported via `Intl.DateTimeFormat` where you can be sure that it works across different languages.

Hope this helps!

**Comment by nkwwk** [comment]
Thanks for the reply, @amannn. However, escape wording should still be implemented. Reference to most common date formatters like moment.js, date-fns, and day.js provided the escape wording. That allows for using unique wording inside the date, for example, '25 of December 2023', which is covered with something like 'dd [of] MMMM yyyy.' Currently, we could use the message to store the format to convert the date, but I suggest extending the ICU format to support the escape wording to make it look better.
